apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -311,12 +313,64 @@ fn validate_field(
         });
         return errors;
     };
-    let url_coordinate = connect_directive_url_coordinate(
-        connect_directive_name,
-        http_verb,
-        object_name,
-        &field.name,
-    );
+    let http_methods: Vec<_> = http_arg
+        .iter()
+        .filter(|(method, _)| {
+            [
+                CONNECT_HTTP_ARGUMENT_GET_METHOD_NAME,
+                CONNECT_HTTP_ARGUMENT_POST_METHOD_NAME,
+                CONNECT_HTTP_ARGUMENT_PUT_METHOD_NAME,
+                CONNECT_HTTP_ARGUMENT_PATCH_METHOD_NAME,
+                CONNECT_HTTP_ARGUMENT_DELETE_METHOD_NAME,
+            ]
+            .contains(method)
+        })
+        .collect();
+    if http_methods.len() > 1 {
+        errors.push(Message {
+            code: Code::MultipleHttpMethods,
+            message: format!(
+                "{coordinate} cannot specify more than one HTTP method.",
+                coordinate = connect_directive_http_coordinate(
+                    connect_directive_name,
+                    object_name,
+                    &field.name
+                ),
+            ),
+            locations: http_methods
+                .iter()
+                .flat_map(|(_, node)| Location::from_node(node.location(), source_map).into_iter())
+                .collect(),
+        });
+    } else if http_methods.is_empty() {
+        errors.push(Message {
+            code: Code::MissingHttpMethod,
+            message: format!(
+                "{coordinate} must specify an HTTP method.",
+                coordinate = connect_directive_http_coordinate(
+                    connect_directive_name,
+                    object_name,
+                    &field.name
+                ),
+            ),
+            locations: Location::from_node(http_arg_location, source_map)
+                .into_iter()
+                .collect(),
+        });
+    }
+    let http_arg_url = if let Some((http_method, url)) = http_methods.first() {
+        Some((
+            url,
+            connect_directive_url_coordinate(
+                connect_directive_name,
+                http_method,
+                object_name,
+                &field.name,
+            ),
+        ))
+    } else {
+        None
+    };
```

Comment:
```suggestion
    let http_arg_url = http_methods.first().map(|(http_method, url)| {
        (
            url,
            connect_directive_url_coordinate(
                connect_directive_name,
                http_method,
                object_name,
                &field.name,
            ),
        )
    });
```

Reply:
Nice, this evolved through a couple iterations and I somehow missed this simplification. I'll update it.